cloud/cloud_env.cc

```diff
@@ -89,7 +89,10 @@ CloudEnv::CloudEnv(const CloudEnvOptions& options, Env* base,
                    const std::shared_ptr<Logger>& logger)
     : cloud_env_options(options), base_env_(base), info_log_(logger) {}
 
-CloudEnv::~CloudEnv() {}
+CloudEnv::~CloudEnv() {
+  cloud_env_options.cloud_log_controller.reset();
+  cloud_env_options.storage_provider.reset();
```

Comment:
this should automatically happen right?

Reply:
It will happen automatically, but not until after the CloudEnv is destroyed.  By doing it here, the CloudEnv still exists.  It is a safety check that may or may not help anything.